use-shopping-cart/src/reducers.js

```diff
@@ -139,6 +142,18 @@ export function cartValuesReducer(state, action) {
     case 'clear-cart':
       return cartValuesInitialState
 
+    case 'load-cart':
+      if (!action.shouldMerge) state = { ...cartValuesInitialState }
+
+      for (const sku in action.cartDetails) {
+        const entry = action.cartDetails[sku]
+        if (action.filter && !action.filter(entry))
+          continue
```

Comment:
I don't think we'll be needing this part since we have the utility function for filtering.